grammar.js

```diff
@@ -250,6 +250,9 @@ module.exports = grammar({
       $._expression,
       $.type_declaration,
     ],
+    [
```

Comment:
Found another conflict when making the params in the parenthesis optional, which I think is one of the edge cases you mentioned.